dash/dash.py

```diff
@@ -2076,3 +2170,137 @@ def verify_url_part(served_part, url_part, part_name):
                     extra_files.append(path)
 
         self.server.run(host=host, port=port, debug=debug, **flask_run_options)
+
+    def _import_layouts_from_pages(self):
+        walk_dir = self.config.pages_folder
+        pages = "pages" if self.pages_folder == "" else self.pages_folder
+
+        for (root, _, files) in os.walk(walk_dir):
+            for file in files:
+                if file.startswith("_") or not file.endswith(".py"):
+                    continue
+                with open(os.path.join(root, file)) as f:
+                    content = f.read()
+                    if "register_page" not in content:
+                        continue
+
+                page_filename = os.path.join(root, file).replace("\\", "/")
+                _, _, page_filename = page_filename.partition(pages + "/")
+                page_filename = page_filename.replace(".py", "").replace("/", ".")
+                module_name = ".".join([pages, page_filename])
+                page_module = importlib.import_module(module_name)
+
+                self.page_registry = _pages.PAGE_REGISTRY.copy()
```

Comment:
Can you say more about why we're copying `PAGE_REGISTRY` into the app instance instead of just continuing to use the one in `_pages`? Note that `.copy()` is shallow, so the assignment you're doing below is modifying the original anyway.

If we want `app.page_registry` to be available for users we could do that (`self.page_registry = _pages.PAGE_REGISTRY` once in the constructor) but even that I don't really see the purpose, since we also have `dash.page_registry`. And keeping the copy if we don't need it is a risk of bugs if there's ever a way the two copies could get out of sync.

Reply:
I thought a `self.page_registry` was required, but it's unnecessary.  It's much better without it.